app/config/security.yml

```diff
@@ -29,10 +39,10 @@ security:
                 - ROLE_SONATA_ADMIN_DEMO_FOO_LIST
                 - ROLE_SONATA_ADMIN_DEMO_FOO_VIEW
                 - ROLE_SONATA_ADMIN
-            ROLE_SUPER_ADMIN: [ROLE_SONATA_FOO_ADMIN, ROLE_ALLOWED_TO_SWITCH]
+            ROLE_SUPER_ADMIN: [ROLE_SONATA_FOO_ADMIN, ROLE_ALLOWED_TO_SWITCH, ROLE_API]
 
     access_control:
             - { path: ^%admin_dashboard_base_url%/, roles: ROLE_SUPER_ADMIN }
-
+            - { path: ^/customers/login, roles: ROLE_API }
```

Comment:
You can't require auntentication for login!
See **Be Sure the Login Page Isn't Secure (Redirect Loop!)** in Symfony documentation
http://symfony.com/doc/current/security/form_login_setup.html#be-sure-the-login-page-isn-t-secure-redirect-loop